docs/cookbook/using-docker.md

```diff
@@ -0,0 +1,343 @@
+
+### Creating a `.dockerignore` file
+The `.dockerignore` file prevents specific files from being copied into your final image. This is important to block sensitive or unnecessary files from being incorporated into your image. Simply go through your directory and copy any file or folder name not needed to build your project into your `.dockerignore` file. Note that folder names are followed by a `/`. I'm using Visual Studio Code in this tutorial, so the topmost folder listed won't be in your project if you use a different editor.
+
+<AposCodeBlock>
+
+```bash
+.vscode/
+apos-build/
+badges/
+data/
+node-modules/
+public/apos-frontend/
+public/uploads/
+.dockerignore
+.env
+.eslintignore
+.gitignore
+deploy-test-count
+docker-compose.yaml
+dockerfile
+force-deploy
+local.example.js
+```
+
+<template v-slot:caption>
+  .dockerignore
+</template>
+
+</AposCodeBlock>
+
+### Creating a `docker-compose.yaml` file
+In this guide, we are starting by creating multiple containers and a persistent volume. This is so that we can provide both a MongoDB instance and a place to store uploaded assets. We are going to do this using [Docker Compose](https://docs.docker.com/compose/) and a `docker-compose.yml` file. In the following sections of the tutorial, we will look at removing the extra container and volume by taking advantage of cloud storage and database services. Create this file at the root of your project.
+
+<AposCodeBlock>
+
+```bash
+services:
+  db:
+    image: mongo:4.4.14
+    ports:
+      - "27017:27017"
+    volumes:
+      - /data/db
+  web:
+    build:
+      context: .
+    ports:
+      - "3000:3000"
+    environment:
+      - NODE_ENV
+      - APOS_MONGODB_URI
+      - APOS_CLUSTER_PROCESSES
+    depends_on:
+      - db 
+    volumes:
+      - /srv/www/apostrophe/public/uploads
+```
+
+<template v-slot:caption>
+  docker-compose.yaml
+</template>
+
+</AposCodeBlock>
+
+The spacing in this file is very important. White-space, not tab, indentation indicates that a particular line is nested within the object passed on the line above it. Walking through this file, it starts with `services:`. From the indentation, we can see that we are creating two services - a `db:` container and a `web:` container.  Much like our `Dockerfile`, within the `db:` we start by specifying an image to run. In this case, it is the `mongo:4.4.14` official image for running MongoDB v4.4.14. Other images can be found in the docker library GitHub repo [README](https://github.com/docker-library/docs/blob/master/mongo/README.md#supported-tags-and-respective-dockerfile-links). You should use the version that mirrors your development environment.
+
+Next, we are specifying that the database should communicate over port `27017`. This is the port typically used by Apostrophe but might need to be altered to match your environment.
```

Comment:
"You might need to alter it to avoid conflict with mongodb running in your development environment."

Actually... you WILL need to alter it, in a typical setup where you've followed the rest of our HOWTOs, if you try this docker-compose file locally.

You could change the host (rather than container) port to 27018, so the container-to-container URI doesn't have to change, but there's no conflict with mongodb running on the host if you want to connect to this container's mongodb using the mongo CLI tools on the host.

Reply:
Yeah - I am not able to work this out.

Reply:
I think the simplest resolution is to consistently use 27018 for mongodb within this docker recipe. Don't try to get clever and map 27018 to 27017 here and there, just use 27018 period. That will be fine as long as you set APOS_MONGODB_URI accordingly.

Reply:
(This is also a good proof that you're actually talking to the dockerized mongo and not to mongo on the host.)

Reply:
Changed.